packages/graphql/src/schema/scalars/Duration.ts

```diff
@@ -0,0 +1,147 @@
+        monthUnit = 0,
+        dayUnit = 0,
+        hourUnit = 0,
+        minuteUnit = 0,
+        secondUnit = 0,
+        // PnW
+        weekUnit = 0,
+        // PYYYYMMDDTHHMMSS | PYYYY-MM-DDTHH:MM:SS
+        yearDT = 0,
+        monthDT = 0,
+        dayDT = 0,
+        hourDT = 0,
+        minuteDT = 0,
+        secondDT = 0,
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    } = match.groups!;
+
+    // NOTE: xUnit and xDT cannot both be nonzero by construction => (xUnit + xDT) = xUnit | xDT | 0
+    const years = +yearUnit + +yearDT;
+    const months = +monthUnit + +monthDT;
+    const weeks = +weekUnit;
+    const days = +dayUnit + +dayDT;
+    const hours = +hourUnit + +hourDT;
+    const minutes = +minuteUnit + +minuteDT;
+    const seconds = +secondUnit + +secondDT;
+
+    // Splits a component into a whole part and remainder
+    const splitComponent = (component: number): [number, number] => [
+        Math.trunc(component),
+        +(component % 1).toPrecision(9),
+    ];
+
+    // Calculate months based off of months and years
+    const [wholeMonths, remainderMonths] = splitComponent(months + years * MONTHS_PER_YEAR);
+
+    // Calculate days based off of days, weeks, and remainder of months
+    const [wholeDays, remainderDays] = splitComponent(days + weeks * DAYS_PER_WEEK + remainderMonths * DAYS_PER_MONTH);
+
+    // Calculate seconds based off of remainder of days, hours, minutes, and seconds
+    const splitHoursInSeconds = splitComponent((hours + remainderDays * HOURS_PER_DAY) * SECONDS_PER_HOUR);
+    const splitMinutesInSeconds = splitComponent(minutes * SECONDS_PER_MINUTE);
+    const splitSeconds = splitComponent(seconds);
+    // Total seconds by adding splits of hour minute second
+    const [wholeSeconds, remainderSeconds] = splitHoursInSeconds.map(
+        (p, i) => p + splitMinutesInSeconds[i] + splitSeconds[i]
+    );
+
+    // Calculate nanoseconds based off of remainder of seconds
+    const wholeNanoseconds = +remainderSeconds.toFixed(9) * 1000000000;
+
+    // Whether total duration is negative
+    const coefficient = negated ? -1 : 1;
+    // coefficient of duration and % may negate zero: converts -0 -> 0
+    const unsignZero = (a: number) => (Object.is(a, -0) ? 0 : a);
+
+    return {
+        months: unsignZero(coefficient * wholeMonths),
+        days: unsignZero(coefficient * wholeDays),
+        seconds: unsignZero(coefficient * wholeSeconds),
+        nanoseconds: unsignZero(coefficient * wholeNanoseconds),
+    };
+};
+
+const parse = (value: any) => {
+    const { months, days, seconds, nanoseconds } = parseDuration(value);
+
+    return new neo4j.types.Duration(months, days, seconds, nanoseconds);
+};
+
+export default new GraphQLScalarType({
+    name: "Duration",
+    description: "A duration, represented as an ISO 8601 duration string",
+    serialize: (value: any) => {
+        if (typeof value !== "string" && !(value instanceof neo4j.types.Duration)) {
```

Comment:
Is this meant to be `||`?

Just trying to wrap my head around this, because after this `if` statement, `toString()` is called on the object, I'm guessing assuming that it's a `Duration` type?

Reply:
This is more of a logical rule than anything, but saying it aloud it does read better. Fixed in https://github.com/neo4j/graphql/pull/459/commits/4663bdfd62c04068f4c840c547847519eab1830c.

Does this make a little more sense? It also gets rid of a regex match by assuming the `Duration.toString()` is a valid duration string which I think is a safe assumption.

Reply:
Definitely makes more sense, and definitely safe to get rid of that RegEx check in my opinion! 🙂